tools/intg-release-migrator/tools/importUtils.ts

```diff
@@ -118,15 +125,29 @@ export async function replaceImports(baseDir: string): Promise<void> {
 
   await Promise.all(
     Object.entries(pathsToPatchImports).map(async ([pathToFile, count]) => {
-      const scriptData = await fs.readFile(pathToFile, 'utf8');
+      const newPathToFile =
+        pathToFile.endsWith('.js') && !pathToFile.includes('/cjs/') && mainIsCJs
+          ? pathToFile.replace(/\.js$/, '.mjs')
+          : pathToFile;
+
+      if (handledFiles.has(newPathToFile)) return;
+      handledFiles.add(newPathToFile);
+
+      const initialContent = await fs.readFile(pathToFile, 'utf8');
 
-      const dataToWrite = scriptData
-        .replace(regexpES, `from '${newName}/$2';`)
-        .replace(regexpCJS, `require("${newName}/$2")`);
+      const newContent = initialContent
+        .replace(shallowRegexpES, `from '${newName}/$2/index.mjs';`)
+        .replace(deepRegexpES, `from '${newName}/$2.mjs';`)
+        .replace(regexpCJS, `require("${newName}/$2")`)
+        .replace(relativeRegexpEs, `from './$1.mjs';`);
```

Comment:
I'm not sure, we should patch imports like that... maybe better solution is to create correct `exports` field in our package.json?